In [23]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [24]:
data = pd.read_csv("C:/Users/kenneth/OneDrive/Desktop/dev/Python Projects/Unified Mentor Internship/Data Visualization of Bird Strikes between 2000 and 2011/Bird Strikes data.xlsx - Bird Strikes.csv")
data

Record ID Aircraft: Type                    Airport: Name Altitude bin  \
0         202152       Airplane                     LAGUARDIA NY    > 1000 ft   
1         208159       Airplane      DALLAS/FORT WORTH INTL ARPT    < 1000 ft   
2         207601       Airplane                LAKEFRONT AIRPORT    < 1000 ft   
3         215953       Airplane              SEATTLE-TACOMA INTL    < 1000 ft   
4         219878       Airplane                     NORFOLK INTL    < 1000 ft   
...          ...            ...                              ...          ...   
25553     321151       Airplane                REDDING MUNICIPAL    > 1000 ft   
25554     319677       Airplane                     ORLANDO INTL    < 1000 ft   
25555     319680            NaN                              NaN          NaN   
25556     319679       Airplane  DETROIT METRO WAYNE COUNTY ARPT    < 1000 ft   
25557     319593       Airplane     ABRAHAM LINCOLN CAPITAL ARPT    < 1000 ft   

      Aircraft: Make/Model Wildlife: Number struck  \
0                B-737-400                Over 100   
1                    MD-80                Over 100   
2                    C-500                Over 100   
3                B-737-400                Over 100   
4             CL-RJ100/200                Over 100   
...                    ...                     ...   
25553              EMB-120                       1   
25554                A-321                       1   
25555               EC-135                     NaN   
25556            B-757-200                       1   
25557            B-737-400                       1   

       Wildlife: Number Struck Actual Effect: Impact to flight     FlightDate  \
0                                 859         Engine Shut Down  11/23/00 0:00   
1                                 424                      NaN   7/25/01 0:00   
2                                 261                      NaN   9/14/01 0:00   
3                                 806    Precautionary Landing    9/5/02 0:00   
4                                 942                      NaN   6/23/03 0:00   
...                               ...                      ...            ...   
25553                               1                      NaN  12/30/11 0:00   
25554                               1                      NaN  12/30/11 0:00   
25555                               1                      NaN            NaN   
25556                               1                      NaN  12/31/11 0:00   
25557                               1                      NaN  12/31/11 0:00   

      Effect: Indicated Damage  ... Remains of wildlife sent to Smithsonian  \
0                Caused damage  ...                                   False   
1                Caused damage  ...                                   False   
2                    No damage  ...                                   False   
3                    No damage  ...                                   False   
4                    No damage  ...                                   False   
...                        ...  ...                                     ...   
25553                No damage  ...                                   False   
25554                No damage  ...                                   False   
25555                No damage  ...                                   False   
25556                No damage  ...                                   False   
25557            Caused damage  ...                                   False   

                                                 Remarks Wildlife: Size  \
0      FLT 753. PILOT REPTD A HUNDRED BIRDS ON UNKN T...         Medium   
1      102 CARCASSES FOUND. 1 LDG LIGHT ON NOSE GEAR ...          Small   
2      FLEW UNDER A VERY LARGE FLOCK OF BIRDS OVER AP...          Small   
3      NOTAM WARNING. 26 BIRDS HIT THE A/C, FORCING A...          Small   
4                                          NO DMG REPTD.          Small   
...          

In [25]:
data.shape

(25558, 26)

25558 rows and 26 columns

In [26]:
#to determine the number of dimension of the array
data.ndim

2

In [27]:
data.describe()

Record ID  Wildlife: Number Struck Actual  Number of people injured
count   25558.000000                    25558.000000              25558.000000
mean   253916.085609                        2.691525                  0.001056
std     38510.453382                       12.793975                  0.050420
min      1195.000000                        1.000000                  0.000000
25%    225783.750000                        1.000000                  0.000000
50%    248749.000000                        1.000000                  0.000000
75%    269168.750000                        1.000000                  0.000000
max    321909.000000                      942.000000                  6.000000

In [28]:
data.isnull().sum()

Record ID                                      0
Aircraft: Type                               129
Airport: Name                                129
Altitude bin                                 129
Aircraft: Make/Model                           0
Wildlife: Number struck                      129
Wildlife: Number Struck Actual                 0
Effect: Impact to flight                   23480
FlightDate                                   129
Effect: Indicated Damage                       0
Aircraft: Number of engines?                 267
Aircraft: Airline/Operator                   129
Origin State                                 449
When: Phase of flight                        129
Conditions: Precipitation                  23543
Remains of wildlife collected?                 0
Remains of wildlife sent to Smithsonian        0
Remarks                                     4771
Wildlife: Size                               129
Conditions: Sky                                0
Wildlife: Species   

In [29]:
#checking if remarks attribute is relevant or not. Since there are many null values in remarks we will remove this attribute if its not relevant
data['Remarks'].unique()

array(['FLT 753. PILOT REPTD A HUNDRED BIRDS ON UNKN TYPE. #1 ENG WAS SHUT DOWN AND DIVERTED TO EWR. SLIGHT VIBRATION. A/C WAS OUT OF SVC FOR REPAIRS TO COWLING, FAN DUCT ACCOUSTIC PANEL. INGESTION. DENTED FAN BLADE #26 IN #1 ENG. HEAVY BLOOD STAINS ON L WINGTIP',
       '102 CARCASSES FOUND. 1 LDG LIGHT ON NOSE GEAR WAS DAMAGED AND REPLACED.',
       'FLEW UNDER A VERY LARGE FLOCK OF BIRDS OVER APCH END OF RWY. NO DMG. JUST A LOT OF BIRD DROPPINGS ON WINDSCREEN.',
       ...,
       'STRUCK BIRD ON RT FRONT DURING T/O. BIRD REPTD AS BROWN/WHITE. TWY.',
       'PILOTS REPORT STRIKING UNKNOWN BIRD ON RWY 21L BTWN TWY F & J. NO REMAINS FOUND ON RWY OR ON A/C. NO DMG TO A/C.',
       'HIT CENTER OF RADOME, CAVING IN ABOUT 12". RADOME WAS REPLACED. CARCASS FOUND IN SAFETY ARA ON RT SIDE OF RWY 22 AT INTXN OF RWY 18/36.'],
      dtype=object)

In [30]:
#checking unique values in 'Origin of State'
data['Origin State'].unique()

array(['New York', 'Texas', 'Louisiana', 'Washington', 'Virginia', nan,
       'Delaware', 'DC', 'Georgia', 'Florida', 'California', 'Illinois',
       'Connecticut', 'Missouri', 'Rhode Island', 'Hawaii', 'Arizona',
       'Tennessee', 'South Carolina', 'South Dakota', 'New Jersey',
       'Colorado', 'Minnesota', 'Alabama', 'Ohio', 'Wisconsin',
       'Michigan', 'Massachusetts', 'Alaska', 'North Carolina',
       'Kentucky', 'Indiana', 'Oregon', 'Pennsylvania', 'New Hampshire',
       'Arkansas', 'Nevada', 'Mississippi', 'Maryland', 'Maine', 'Quebec',
       'Idaho', 'British Columbia', 'Utah', 'Nebraska', 'Iowa',
       'New Mexico', 'West Virginia', 'Oklahoma', 'North Dakota',
       'Vermont', 'Wyoming', 'Kansas', 'Prince Edward Island', 'Montana',
       'Puerto Rico', 'Ontario', 'Virgin Islands',
       'Newfoundland and Labrador', 'Alberta', 'Saskatchewan'],
      dtype=object)

In [31]:
#checking unique values in 'Phase of flight'
data['When: Phase of flight'].unique()

array(['Climb', 'Landing Roll', 'Approach', 'Take-off run', 'Descent',
       nan, 'Taxi', 'Parked'], dtype=object)

In [32]:
#checking unique values of aircraft type
data['Aircraft: Type'].unique()

array(['Airplane', nan], dtype=object)

In [33]:
#getting all the column names
data.columns

Index(['Record ID', 'Aircraft: Type', 'Airport: Name', 'Altitude bin',
       'Aircraft: Make/Model', 'Wildlife: Number struck',
       'Wildlife: Number Struck Actual', 'Effect: Impact to flight',
       'FlightDate', 'Effect: Indicated Damage',
       'Aircraft: Number of engines?', 'Aircraft: Airline/Operator',
       'Origin State', 'When: Phase of flight', 'Conditions: Precipitation',
       'Remains of wildlife collected?',
       'Remains of wildlife sent to Smithsonian', 'Remarks', 'Wildlife: Size',
       'Conditions: Sky', 'Wildlife: Species',
       'Pilot warned of birds or wildlife?', 'Cost: Total $',
       'Feet above ground', 'Number of people injured', 'Is Aircraft Large?'],
      dtype='object')

In [34]:
#getting the size of the data
data.size

664508

rows * columns

In [35]:
#getting details like number of non null values and data type of each column
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25558 entries, 0 to 25557
Data columns (total 26 columns):
 #   Column                                   Non-Null Count  Dtype 
---  ------                                   --------------  ----- 
 0   Record ID                                25558 non-null  int64 
 1   Aircraft: Type                           25429 non-null  object
 2   Airport: Name                            25429 non-null  object
 3   Altitude bin                             25429 non-null  object
 4   Aircraft: Make/Model                     25558 non-null  object
 5   Wildlife: Number struck                  25429 non-null  object
 6   Wildlife: Number Struck Actual           25558 non-null  int64 
 7   Effect: Impact to flight                 2078 non-null   object
 8   FlightDate                               25429 non-null  object
 9   Effect: Indicated Damage                 25558 non-null  object
 10  Aircraft: Number of engines?             25291 non-null  o

In [36]:
#making a copy of the data
data_copy = data.copy()
#making changes on the copied data

In [37]:
data_copy

Record ID Aircraft: Type                    Airport: Name Altitude bin  \
0         202152       Airplane                     LAGUARDIA NY    > 1000 ft   
1         208159       Airplane      DALLAS/FORT WORTH INTL ARPT    < 1000 ft   
2         207601       Airplane                LAKEFRONT AIRPORT    < 1000 ft   
3         215953       Airplane              SEATTLE-TACOMA INTL    < 1000 ft   
4         219878       Airplane                     NORFOLK INTL    < 1000 ft   
...          ...            ...                              ...          ...   
25553     321151       Airplane                REDDING MUNICIPAL    > 1000 ft   
25554     319677       Airplane                     ORLANDO INTL    < 1000 ft   
25555     319680            NaN                              NaN          NaN   
25556     319679       Airplane  DETROIT METRO WAYNE COUNTY ARPT    < 1000 ft   
25557     319593       Airplane     ABRAHAM LINCOLN CAPITAL ARPT    < 1000 ft   

      Aircraft: Make/Model Wildlife: Number struck  \
0                B-737-400                Over 100   
1                    MD-80                Over 100   
2                    C-500                Over 100   
3                B-737-400                Over 100   
4             CL-RJ100/200                Over 100   
...                    ...                     ...   
25553              EMB-120                       1   
25554                A-321                       1   
25555               EC-135                     NaN   
25556            B-757-200                       1   
25557            B-737-400                       1   

       Wildlife: Number Struck Actual Effect: Impact to flight     FlightDate  \
0                                 859         Engine Shut Down  11/23/00 0:00   
1                                 424                      NaN   7/25/01 0:00   
2                                 261                      NaN   9/14/01 0:00   
3                                 806    Precautionary Landing    9/5/02 0:00   
4                                 942                      NaN   6/23/03 0:00   
...                               ...                      ...            ...   
25553                               1                      NaN  12/30/11 0:00   
25554                               1                      NaN  12/30/11 0:00   
25555                               1                      NaN            NaN   
25556                               1                      NaN  12/31/11 0:00   
25557                               1                      NaN  12/31/11 0:00   

      Effect: Indicated Damage  ... Remains of wildlife sent to Smithsonian  \
0                Caused damage  ...                                   False   
1                Caused damage  ...                                   False   
2                    No damage  ...                                   False   
3                    No damage  ...                                   False   
4                    No damage  ...                                   False   
...                        ...  ...                                     ...   
25553                No damage  ...                                   False   
25554                No damage  ...                                   False   
25555                No damage  ...                                   False   
25556                No damage  ...                                   False   
25557            Caused damage  ...                                   False   

                                                 Remarks Wildlife: Size  \
0      FLT 753. PILOT REPTD A HUNDRED BIRDS ON UNKN T...         Medium   
1      102 CARCASSES FOUND. 1 LDG LIGHT ON NOSE GEAR ...          Small   
2      FLEW UNDER A VERY LARGE FLOCK OF BIRDS OVER AP...          Small   
3      NOTAM WARNING. 26 BIRDS HIT THE A/C, FORCING A...          Small   
4                                          NO DMG REPTD.          Small   
...          

In [38]:
data_copy.isnull().sum()

Record ID                                      0
Aircraft: Type                               129
Airport: Name                                129
Altitude bin                                 129
Aircraft: Make/Model                           0
Wildlife: Number struck                      129
Wildlife: Number Struck Actual                 0
Effect: Impact to flight                   23480
FlightDate                                   129
Effect: Indicated Damage                       0
Aircraft: Number of engines?                 267
Aircraft: Airline/Operator                   129
Origin State                                 449
When: Phase of flight                        129
Conditions: Precipitation                  23543
Remains of wildlife collected?                 0
Remains of wildlife sent to Smithsonian        0
Remarks                                     4771
Wildlife: Size                               129
Conditions: Sky                                0
Wildlife: Species   

In [39]:
#dropping the remarks column
data_copy.drop(labels = 'Remarks',axis = 1,inplace = True)

In [40]:
#dropping null values from the neccessary columns except the Remarks column
data_copy.dropna(subset=['Aircraft: Type','Airport: Name','Altitude bin','Wildlife: Number struck','Effect: Impact to flight','FlightDate','Aircraft: Number of engines?','Aircraft: Airline/Operator','Origin State','When: Phase of flight','Conditions: Precipitation','Wildlife: Size','Pilot warned of birds or wildlife?','Feet above ground','Is Aircraft Large?'], inplace = True)

In [41]:
data_copy.isnull().sum()

Record ID                                  0
Aircraft: Type                             0
Airport: Name                              0
Altitude bin                               0
Aircraft: Make/Model                       0
Wildlife: Number struck                    0
Wildlife: Number Struck Actual             0
Effect: Impact to flight                   0
FlightDate                                 0
Effect: Indicated Damage                   0
Aircraft: Number of engines?               0
Aircraft: Airline/Operator                 0
Origin State                               0
When: Phase of flight                      0
Conditions: Precipitation                  0
Remains of wildlife collected?             0
Remains of wildlife sent to Smithsonian    0
Wildlife: Size                             0
Conditions: Sky                            0
Wildlife: Species                          0
Pilot warned of birds or wildlife?         0
Cost: Total $                              0
Feet above

In [42]:
data_copy.describe()

Record ID  Wildlife: Number Struck Actual  Number of people injured
count     187.000000                      187.000000                     187.0
mean   244446.983957                        7.754011                       0.0
std     30922.201557                       17.876922                       0.0
min    200097.000000                        1.000000                       0.0
25%    221812.500000                        1.000000                       0.0
50%    237720.000000                        1.000000                       0.0
75%    260219.500000                        6.000000                       0.0
max    319260.000000                       97.000000                       0.0

count : this means the in the first column there are 187 non null values in the second column there are again 187 non null values and so on

std : standard deviation - which measures the amount of variation in a set of values

25% 50% and 75% are percentile and not percentage
25% of the values are less than or equal to 221812.5

In [44]:
data_copy

Record ID Aircraft: Type                           Airport: Name  \
190       206599       Airplane                              LOGAN INTL   
210       208033       Airplane                MINNEAPOLIS-ST PAUL INTL   
222       205109       Airplane  CINCINNATI/NORTHERN KENTUCKY INTL ARPT   
260       209544       Airplane                      METRO OAKLAND INTL   
273       206608       Airplane                            ATLANTA INTL   
...          ...            ...                                     ...   
24550     315222       Airplane              WILEY POST-WILL ROGERS MEM   
24784     316086       Airplane                CRAWFORDSVILLE MUNI ARPT   
25078     317500       Airplane      CROOKSTON MUNI ARPT-KIRKWOOD FIELD   
25372     319260       Airplane                        KANSAS CITY INTL   
25417     318851       Airplane             CHARLOTTE/DOUGLAS INTL ARPT   

      Altitude bin Aircraft: Make/Model Wildlife: Number struck  \
190      < 1000 ft             SAAB-340                 2 to 10   
210      > 1000 ft             SAAB-340                 2 to 10   
222      < 1000 ft         CL-RJ100/200                 2 to 10   
260      > 1000 ft            B-737-400                 2 to 10   
273      < 1000 ft                A-320                 2 to 10   
...            ...                  ...                     ...   
24550    < 1000 ft           LEARJET-31                       1   
24784    < 1000 ft      CIRRUS SR 20/22                       1   
25078    > 1000 ft                C-172                       1   
25372    < 1000 ft            B-737-300                       1   
25417    < 1000 ft              EMB-170                       1   

       Wildlife: Number Struck Actual Effect: Impact to flight     FlightDate  \
190                                 5         Aborted Take-off  10/16/00 0:00   
210                                 3    Precautionary Landing  11/13/00 0:00   
222                                 5    Precautionary Landing  12/13/00 0:00   
260                                 5    Precautionary Landing    3/6/01 0:00   
273                                 9    Precautionary Landing   4/13/01 0:00   
...                               ...                      ...            ...   
24550                               1         Aborted Take-off   8/22/11 0:00   
24784                               1    Precautionary Landing   9/14/11 0:00   
25078                               1    Precautionary Landing   10/8/11 0:00   
25372                               1    Precautionary Landing   11/7/11 0:00   
25417                               1         Aborted Take-off  11/16/11 0:00   

      Effect: Indicated Damage  ... Remains of wildlife collected?  \
190                  No damage  ...                          False   
210              Caused damage  ...                          False   
222              Caused damage  ...                          False   
260              Caused damage  ...                          False   
273              Caused damage  ...                          False   
...                        ...  ...                            ...   
24550            Caused damage  ...                           True   
24784                No damage  ...                          False   
25078                No damage  ...                          False   
25372                No damage  ...                           True   
25417            Caused damage  ...                           True   

      Remains of wildlife sent to Smithsonian Wildlife: Size Conditions: Sky  \
190                                     False          Small        Overcast   
210                                     False          Large        Overcast   
222                                     False         Medium        Overcast   
260                                     False         Medium      Some Cloud   
273                                     False         Medium      Some Cloud   
...    

In [45]:
#exporting the data into an excel sheet
data_copy.to_excel(r'C:/Users/kenneth/OneDrive/Desktop/dev/Python Projects/Unified Mentor Internship/Data Visualization of Bird Strikes between 2000 and 2011/Bird Strikes data_cleaned.xlsx',index = False)

In [2]:
pip install nbconvert

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.3.2 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


cleaned data without null values is stored in the above location